flatpak-proxy.c

```diff
@@ -1593,16 +1592,16 @@ parse_header (Buffer *buffer, guint32 serial_offset, guint32 reply_serial_offset
       return NULL;
     }
 
-  if (serial_offset > 0)
+  if (translate_serial)
     {
-      header->serial += serial_offset;
+      header->serial = header->serial * 2 + 1;
       write_uint32 (header, &buffer->data[8], header->serial);
     }
 
-  if (reply_serial_offset > 0 &&
-      header->has_reply_serial &&
-      header->reply_serial > hello_serial + reply_serial_offset)
```

Comment:
Is this just a plausibility check?

We need to know if the `hello_serial` is stored with the transformation applied or not. Then we can probably do something equivalent to the existing check.

Reply:
Since we don't do any translations anymore, this should be fine.